std-bits/base/src/main/java/org/enso/base/Text_Utils.java

```diff
@@ -35,6 +35,30 @@ public static String substring(String string, int from, int to) {
     return string.substring(from, to);
   }
 
+  /**
+   * Checks if the string has leading or trailing whitespace.
+   *
+   * @param s the string to check
+   * @return whether the string has leading or trailing whitespace
+   */
+  public static boolean has_leading_trailing_whitespace(String s) {
+    if (s == null && s.isEmpty()) {
```

Comment:
```suggestion
    if (s == null || s.isEmpty()) {
```

looks like a typo, the current condition will actually fail with NPE if `s == null`